xbmc/URL.cpp

```diff
@@ -253,35 +253,29 @@ void CURL::Parse(const std::string& strURL1)
     }
   }
 
+  std::string strHostNameAndPort = strURL.substr(iPos, (iSlash == std::string::npos) ? iEnd - iPos : iSlash - iPos);
+  // check for IPv6 numerical representation inside []
+  size_t iBrk = strHostNameAndPort.rfind("]");
+  if (iBrk < iEnd && iBrk != std::string::npos && strHostNameAndPort.find("[") == 0)
+  {
+    m_strHostName = strHostNameAndPort.substr(1, iBrk-1);
+    strHostNameAndPort.erase(0, iBrk+1);
```

Comment:
ok, clearly iEnd has also no meaning after creating substr and it should be replaced
with strHostNameAndPort.size().
(although thinking a bit more, the "iBrk < iEnd" or "iBrk < strHostNameAndPort.size()" respectively makes no sense. iBrk will be either "npos", "iBrk < strHostNameAndPort.size()" otherwise. so I will remove that from if().

then the substr() and erase should be fine as if the if() block is entered, we know '[' is at start (0), we copy from 1 until the closing ']'. position  of ']' is maximum at string.size()-1 that means iBrk-1 can represent max "string.size()-2" and by copying this number of bytes (even from 1, not 0) we never reach behind strings size.

then erasing from beginning iBrk+1 bytes - doing same assumption of limit case (URI contains no more than host definition inside [], iBrk+1 will represent exactly "string.size()" bytes. removing those from position 0 puts as to index "string.size()-1" (so basically we make string empty. 

if we see eye to eye on this I will remove the useless "iBrk < iEnd" part.